pkg/apis/pipeline/v1alpha1/pipeline_validation.go

```diff
@@ -138,6 +138,13 @@ func (ps *PipelineSpec) Validate(ctx context.Context) *apis.FieldError {
 	// Names cannot be duplicated
 	taskNames := map[string]struct{}{}
 	for i, t := range ps.Tasks {
+		if errs := validation.IsDNS1123Label(t.Name); len(errs) > 0 {
+			return &apis.FieldError{
+				Message: fmt.Sprintf("invalid value %q", t.Name),
+				Paths:   []string{fmt.Sprintf("spec.tasks[%d].name", i)},
+				Details: "Pipeline Task step name must be a valid DNS Label, For more info refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names",
```

Comment:
Do we want the error message to be consistent with Task?

```
The Task "FooTask" is invalid: metadata.name: Invalid value: "FooTask": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

Remove `step` after `Pipeline Task`:

```suggestion
				Details: "Pipeline Task name must be a valid DNS Label. For more info refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names",
```

Reply:
Seeing there was another push here, but I am not seeing an update to the error message.

My thought is to make it so the error message here is consistent with what is shown for a Task that a user tries to create the violates naming conventions.